pymeasure/instruments/activetechnologies/AWG401x.py

```diff
@@ -0,0 +1,1145 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""This module implements an interface for Active Technologies AWG-401x, both
+for the Arbitrary Waveform Generator (AWG) mode and the Arbitrary Function
+Generator (AFG) mode. The module has been developed from the official
+documentation available on https://www.activetechnologies.it"""
+
+from collections import abc, namedtuple
+
+import numpy as np
+
+import pprint
+
+from pymeasure.adapters.visa import VISAAdapter
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set,\
+    strict_range, joined_validators
+
+
+class Channel_base(Instrument):
+    """Implementation of a base Active Technologies AWG-4000 channel."""
+
+    enabled = Instrument.control(
+        "OUTPut<ch>:STATe?", "OUTPut<ch>:STATe %d",
+        """A boolean property that enables or disables the output for the
+        specified channel.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    polarity = Instrument.control(
+        "OUTPut<ch>:POLarity?", "OUTPut<ch>:POLarity %s",
+        """This property inverts the output waveform relative to its average
+        value: (High Level – Low Level)/2. NORM for normal, INV for inverted
+        """,
+        validator=strict_discrete_set,
+        values=["NORMAL", "NORM", "INVERTED", "INV"],
+        get_process=lambda v: "NORM" if v == 0 else ("INV" if v == 1 else v)
+    )
+
+    delay = Instrument.control(
+        None, None,
```

Comment:
Yes, since the command for the AFG and AWG channel's delay are different i will set in the dedicated class derived from `Channel_base`. e.g., https://github.com/pymeasure/pymeasure/blob/9e0d254463e05259eff6093f21b7c761fc5aba42/pymeasure/instruments/activetechnologies/AWG401x.py#L145-L149